### ISSUE TITLE
list of managed domains not lsited correctly

### ISSUE THREAD

**Issue by rhoerbe** [Emergya/peer#90]
I added the user rhoerbe to a number of domains at http://geant.emergya.es. The domains do not show up in the entity list, but are available in the drop-down when adding a new entity.

**Comment by enriquepablo** [comment]
That is the semantics. In the list of domains you only see the domains that you can manage. In that list you can add members to a domain so they can use them to create entities, but not so that they can manage them.

**Comment by rhoerbe** [comment]
Aha, so "domain owner" and "grantee" are different roles? Then the UI need to be fixed, e.g. by changing menu and page titles from "Domains" to "Manage Domains" (or to go one step further, "Manage FQDNs" - see issue #91). 
However, the semantics of your approach seem to be broken. If there is a fixed "domain owner", then the "manage domain" function is orphaned when staff changes at an organization. The proven RBAC pattern is NOT to have an owner, but a "manager" role (can grant roles) and "user" (can do stuff with data). In a system as simple as PEER, it would even make sense to collapse manager and user into a single role.

BTW, what are the access control rules for domains/FQDNs vs. entities? Does access to a domain imply automatically edit access to the entities?

**Comment by enriquepablo** [comment]
I agree that a manager role would be better that owner. This is somewhat alleviated at the moment by the possibility of changing ownership in the admin site.

Access control to entities is separate from domains, and is defined [here](https://github.com/Emergya/peer/blob/master/peer/entity/security.py), it's quite simple.

**Comment by rhoerbe** [comment]
If I recall correctly there is a 1:1 relationship between domain and entity. What is the rationale to have a separated access management for them?

**Comment by enriquepablo** [comment]
Domains are not really "accessed" as content; they hold no more information than their name, and they cannot be modified. Either you can use a domain to create an entity, or you can't. And there can be more than one entity associated to a single domain.
Entities, on the other hand, have more information associated with them, that can be accessed and edited. Once you create an entity associated to a domain, you manage access to it independently from the users that can use its associated domain to create more entities. To me it seems natural, but I'm probably missing something?

**Comment by rhoerbe** [comment]
For my understanding:
So the only point of a domain is to validate that the user is entitled to use the entity's FQDN. Do I get it right, that you implemented a process where:
1. user A will create a FQDN and proof his entitlement,
2a. User A will add user B to the manage domain team, and user B will add the entity, or
2b. User A will add the entity.

Also please confirm my understanding that it is a rare exception that multiple entities will share a single FQDN.

**Comment by enriquepablo** [comment]
This is correct.
I would only add that it being a rare exception that multiple domains share a single FQDN is not enforced at all by the application, it is a question of users' policies. The app simply allows it.

**Comment by enriquepablo** [comment]
I would further add that in your example, once user B is in the manage team for the domain, she will be able to add any FQDN that is a subdomain of the original, w/o the need to prove ownership.

**Comment by rhoerbe** [comment]
Ah, I missed that now the registration of superdomains is sufficient for entity registration.

My conclusion is that the rights management is too complicated on one side, and too simple OTOH.
Too complicated is, that PEER tries to support separated group memberships for FQDNs and entities. Too simple is the ownership of a domain by a person.

My recommendation would be to collapse domain management and entity groups, and leave it to organizations to manage their internal role segregation outside PEER.
